nilearn/glm/tests/test_thresholding.py

```diff
@@ -34,93 +52,120 @@ def test_fdr():
     assert np.isfinite(fdr_threshold(norm.isf(pvals), 0.1))
 
 
-def test_threshold_stats_img():
-    shape = (9, 10, 11)
-    p = np.prod(shape)
-    data = norm.isf(np.linspace(1.0 / p, 1.0 - 1.0 / p, p)).reshape(shape)
-    alpha = 0.001
-    data[2:4, 5:7, 6:8] = 5.0
-    stat_img = nib.Nifti1Image(data, np.eye(4))
-    mask_img = nib.Nifti1Image(np.ones(shape), np.eye(4))
-
-    # test 1
-    th_map, _ = threshold_stats_img(
-        stat_img, mask_img, alpha, height_control="fpr", cluster_threshold=0
+@pytest.fixture
+def data_norm_isf():
+    p = np.prod(_shape_3d_default())
+    return norm.isf(np.linspace(1.0 / p, 1.0 - 1.0 / p, p)).reshape(
+        _shape_3d_default()
     )
-    vals = get_data(th_map)
-    assert np.sum(vals > 0) == 8
 
-    # test 2: excessive cluster forming threshold
+
+@pytest.fixture
+def img_3d_ones():
+    return Nifti1Image(np.ones(_shape_3d_default()), _affine_eye())
+
+
+def test_threshold_stats_img_no_height_control(
+    data_norm_isf, img_3d_ones, affine_eye
+):
+    data = data_norm_isf
+    data[2:4, 5:7, 6:8] = 5.0
+    stat_img = Nifti1Image(data, affine_eye)
+
+    # excessive cluster forming threshold
     th_map, _ = threshold_stats_img(
         stat_img,
-        mask_img,
+        mask_img=img_3d_ones,
         threshold=100,
         height_control=None,
         cluster_threshold=0,
     )
     vals = get_data(th_map)
     assert np.sum(vals > 0) == 0
 
-    # test 3: excessive size threshold
```

Comment:
Is this one gone ?

Reply:
nope moved below (line 127) in a separate test that have `height_control != None`